async.md

```diff
@@ -0,0 +1,108 @@
+---
+---
+
+# A note on asynchronous code
+
+You might have noticed something in the small test we wrote in the introduction, namely the call to `await`. The Node.js engine is asynchronous, so when a call is made that takes some time to return (such as an API or Database call), rather than wasting time waiting for a response, the engine just processes the code that follows. This can result in issues when you try to use a variable that has not yet been initialized. For example, if you had:
+
+```
+test('hello!', () => {
+  response = await request(app).get('/')
```

Comment:
Did you mean for this `await` to be there? This function is not `async` and it seems like you're trying to demonstrate `response` being `undefined` because `get()` doesn't return anything immediately. If that's the case, remove the `await` here.

Reply:
Oh yeah, I didn't. Will fix 👍